### PR TITLE
build: read user specified defines from file

### PR DESCRIPTION
**Pull request by cruwaller**
User can define specific flags in $HOME/expresslrs.txt or
src/user_defines.txt files to avoid unnecessary file modifications.
This also helps to keep specific regulatory domain and UID in
place.

example content of the file:
-DRegulatory_Domain_EU_868
-DMY_UID=0xBA,0xBE,0xCA,0xFE,0xAC,0xDC

### CODE REVIEW

**Comment by JyeSmith** [comment]
I like this idea.  A couple of things;

- Do we need both expresslrs.txt and user_defines.txt?  Maybe just user_defines.txt?
- Can you add the file to this PR with example/default settings and comment it well for new users.
- Should we add it to gitignore? or is that just asking for trouble when it is modified.

**Comment by cruwaller** [comment]
Thanks! Good points.
I will remove the another file and leave one in the src root and it should be ignored. 
Example file is already on my laptop and missed it from commit. will add it.

File: src/.gitignore [hunk 1]
```diff
@@ -5,3 +5,5 @@
 .vscode/c_cpp_properties.json
 .vscode/launch.json
 __pycache__
+*.pyc
+defines_example.txt
```
Comment (JyeSmith) [inline src/.gitignore]: make this user_defines.txt

Comment (JyeSmith) [reply src/.gitignore]: Rename "user_defines_example.txt" to "user_defines.txt" and include all of the -D options but commented out.  This way new user can clearly see what is available and just needs to uncomment their Reg Domain and enter a UID.

Comment (cruwaller) [reply src/.gitignore]: Yeah, fixed now.

**Pull request merged**